src/session_impl.cpp

```diff
@@ -6865,6 +6950,7 @@ namespace aux {
 			, struct tracker_response const& resp)
 		{
 			TORRENT_UNUSED(tracker_ips);
+			if (!should_log()) return;
```

Comment:
I believe this class (tracker_logger) is instantiated in exactly one place. If instead of instantiating it, you pass in nullptr (or an empty shared_ptr perhaps) I believe it will still work, just not log anything. That way you don't even pay the cost of this object allocation